jevan-de/observer/log_aggregator.py

```diff
@@ -0,0 +1,20 @@
+from typing import List
+from logger import Logger
+from subject import Subject
+
+
+class LogAggregator(Subject):
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._observers: List[Logger] = []
+
+    def register(self, o: Logger) -> None:
```

Comment:
True, there's no type checking yet (but I think they're working on it). In my opinion, in this example, a `LogAggegrator` should only be able to observe `Logger` instances since otherwise it would no longer be a `LogAggregator`. If there where different types of Loggers then they could also inherit from `Logger`.

Reply:
Yeah, I think limiting the type to be logger is a fine decision. Though at that point you might want a more specific abstract class.